libraries/bsg_manycore.cpp

```diff
@@ -1284,6 +1284,59 @@ int hb_mc_manycore_write32(hb_mc_manycore_t *mc, const hb_mc_npa_t *npa, uint32_
         return hb_mc_manycore_write(mc, npa, &v, 4);
 }
 
+/**
+ * Do a 32-bit amoadd to manycore hardware at a given NPA (must be a DRAM address)
+ * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+ * @param[in]  npa    A valid hb_mc_npa_t aligned to a four byte boundary
+ * @param[in]  v      A word value to be added to the NPA
+ * @param[out] vpo    The previous value at the NPA
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_manycore_amoadd(hb_mc_manycore_t *mc, const hb_mc_npa_t *npa, const uint32_t v, uint32_t *vpo)
+{
+        int err;
+        hb_mc_packet_t rqst;
+
+        /* format the request packet */
+        err = hb_mc_manycore_format_request_packet(mc, &rqst.request, npa);
+        if (err != HB_MC_SUCCESS)
+                return err;
+
+        hb_mc_epa_t epa = hb_mc_npa_get_epa(npa);
+
+        // AMO are always 4B
+        size_t sz = 4;
```

Comment:
it's been a while since i've lived in this code base. do we hard code '4' much? maybe we should make a macro 'FOUR' :-D

Reply:
Indeed we do :)
```
int hb_mc_manycore_write32(hb_mc_manycore_t *mc, const hb_mc_npa_t *npa, uint32_t v)
{
        return hb_mc_manycore_write(mc, npa, &v, 4);
}
```

Reply:
haha, well that's very slightly unfortunate, but hey whatever.

Reply:
since it's now embedded in the name of the function that this is a 32-bit op, it seems more OK.